pwa/src/templates/templateParts/download/DownloadTemplate.tsx

```diff
@@ -1,63 +1,82 @@
 import * as React from "react";
 import * as styles from "./DownloadTemplate.module.css";
-
+import clsx from "clsx";
 import { DownloadCard, NotificationPopUp as _NotificationPopUp } from "@conduction/components";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faArrowLeft, faDownload } from "@fortawesome/free-solid-svg-icons";
 import { useTranslation } from "react-i18next";
-import clsx from "clsx";
 import { useGatsbyContext } from "../../../context/gatsby";
+import { TOOLTIP_ID } from "../../../layout/Layout";
 
 interface DownloadTemplateProps {
-  icon: JSX.Element;
-  label: string;
-  sizeKb: string;
+  items: DownloadProps[];
+  backUrl: string;
+  icon?: JSX.Element;
   layoutClassName?: string;
 }
 
-export const DownloadTemplate: React.FC<DownloadTemplateProps> = ({ layoutClassName, icon, label, sizeKb }) => {
+export interface DownloadProps {
+  label: string;
+  type: string;
+  size: string;
+  downloadLink: string;
+}
+
+export const DownloadTemplate: React.FC<DownloadTemplateProps> = ({ items, backUrl, icon, layoutClassName }) => {
   const { t } = useTranslation();
   const { screenSize } = useGatsbyContext();
+  const [isVisible, setIsVisible] = React.useState<any>({});
 
-  const NotificationPopUpController = _NotificationPopUp.controller;
-  const NotificationPopUp = _NotificationPopUp.NotificationPopUp;
+  const toggleVisibility = (label: string) => {
+    setIsVisible({
+      [label]: {
+        value: true,
+      },
+    });
+  };
```

Comment:
I think we can simplify this `isVisible` logic. Right now, we're not really typing it, allowing `any`.

Eventually we're only checking if the `isVisible[label] = true`, what if we simply checked if this label is the current visible one?

Like so:

```
const [isVisible, setIsVisible] = React.useState<string>("");
const toggleVisibility = (label: string) => setIsVisible(label);


// The check can now become
isVisible === (label ?? downloadLink);

// The hide can become
setIsVisible("");
```

Reply:
Iterating on the idea above, we could also use the index of the given array, like so:

```
const [visibleItemIdx, setVisibleItemIdx] = React.useState<number>(-1); // default to -1, which is never available

<DownloadCard
  ... other props
  handleClick={() => setVisibleItemIdx(idx)} // instead of the toggle, simply call the state directly
/>

...

{visibleItemIdx === idx && ( // check once if the pop-up should exist
  <div className={styles.overlay}>
    <NotificationPopUp
      isVisible // do not check here again
      hide={() => setVisibleItemIdx(-1)} // set to -1 to hide everything
      ... nothing changed
    />
  </div>
)}
```